### PR TITLE
[textual inversion] add gradient checkpointing and small fixes.

### PR DESCRIPTION
**Pull request by patil-suraj**
This PR is a continuation of #687 by @Ttl . This PR

- Adds an argument `--gradient_checkpointing` to enable gradient checkpointing. The gradient checkpointing is enabled for the `text_encoder` and `unet`. For this, the `unet` is kept in train mode and as the dropout is 0 it should not affect the results.
- Remove `freeze_params` and instead use `requires_grad_` to disable grads.
- Remove the redundant call to  `accelerator.wait_for_everyone()` after every epoch, it's only needed before saving the pipeline.

### CODE REVIEW

File: examples/textual_inversion/textual_inversion.py [hunk 1]
```diff
@@ -541,9 +541,10 @@ def main():
     unet.to(accelerator.device, dtype=weight_dtype)
     vae.to(accelerator.device, dtype=weight_dtype)
 
-    # Keep vae and unet in eval model as we don't train these
-    vae.eval()
```
Comment (patil-suraj) [inline examples/textual_inversion/textual_inversion.py:545]: `vae` is already in `eval` mode by default, when loaded using `from_pretrained`.

**Comment by patil-suraj** [comment]
Did a trial run, and verified that keeping the `unet` in `train` mode when gradient checkpointing is enabled does not affect the results.

File: examples/textual_inversion/textual_inversion.py [hunk 2]
```diff
@@ -541,9 +541,10 @@ def main():
...
-    unet.eval()
+    # Keep unet in train mode if we are using gradient checkpointing to save memory.
+    # The dropout is 0 so it doesn't matter if we are in eval or train mode.
+    if args.gradient_checkpointing:
+        unet.train()
```
[the remark this answers is not in the document]
Comment (patil-suraj) [reply examples/textual_inversion/textual_inversion.py:547]: It's the `gradient_checkpointing` that saves the memory, but `gradient_checkpointing` is only enabled if the model is in `train` mode. Hence we explicitly put the model here in `train` mode when `args.gradient_checkpointing` is `True`.

[the remark this answers is not in the document]
Comment (pcuenca) [reply examples/textual_inversion/textual_inversion.py:547]: Ah, ok, I misunderstood, thanks! I guess we'll need to be careful when/if we add dropout to the model :)

**Comment by patrickvonplaten** [comment]
Thanks!
